### DISCUSSION TITLE
S3 storage - MinIO addon for Freetz-NG – Any Interest?

### DISCUSSION THREAD

**Discussion by MartMet** [Freetz-NG/freetz-ng#1052] [Allgemein]
Hey everyone,

I recently compiled the MinIO executable for Freetz-NG and tested it on my Fritz!Box 6591. It's working as expected. For those unfamiliar, MinIO is an object storage server that’s compatible with S3 APIs, useful for setting up personal cloud storage or similar services.

A few things I’ve noticed during testing:

The compiled executable is around ~125MB, and I copied it to /tmp for testing purposes.

As a result, RAM is nearly maxed out on my device, so running it this way isn’t ideal for long-term use.
I’m wondering if the size alone would be a dealbreaker for most users, given the RAM and storage limitations on routers like the 6591.

I'm curious to know if anyone would be interested in using MinIO on their Freetz-NG setup as addon. Would this be useful, or does the size and resource consumption make it impractical?

Let me know your thoughts!

Thanks

**Comment by fda77** [comment]
Is it a package like all other to put into make/pkgs/whatever/ ? then its no problem to add it, just create a pull-request
For big files "external" could be used, but is this binary really so huge?
/tmp is ram -> when its full you device will crash

**Comment by MartMet** [comment]
> /tmp is ram -> when its full you device will crash

Its an ugly workaround as i cant run the executeable directly from the mounted HDD. Sidetopic: can the HDD be mounted to allow execution? 

> For big files "external" could be used, but is this binary really so huge?

Yes, even when compiled without any debug info and stripped down the size is about ~90MB

> Is it a package like all other to put into make/pkgs/whatever/ ? then its no problem to add it, just create a pull-request

Actually its just the compiled executable (Intel 80386, statically linked) and its compiled with Go. 
Ive no idea how to create a package.

**Reply by fda77** [reply]
> can the HDD be mounted to allow execution?

there is an option in menuconfig

> 80386

thats only new docsis fritzbox devices
